backend/inspirehep/pidstore/minters/base.py

```diff
@@ -145,7 +145,17 @@ def update(cls, object_uuid, data):
     @classmethod
     def delete(cls, object_uuid, data):
         if "control_number" in data:
-            pid_provider = cls.provider.get(data["control_number"], cls.pid_type)
+            try:
+                pid_provider = cls.provider.get(data["control_number"], cls.pid_type)
+            except PIDDoesNotExistError:
+                if data.get("deleted", False) == True:
+                    cls.mint(object_uuid, data)
```

Comment:
does this fail if there's a texkey/doi/arXiv ID/ORCID/BAI collision with a non-deleted record?

Reply:
This code is in `ControlNumberMinter`. So this will be called only for control_numbers as we only support redirecting control_numbers (texkeys are not redirected but re-claimed)